### ISSUE TITLE
take_along_axis inconsistent with numpy

### ISSUE THREAD

**Issue by cooijmanstim** [google/jax#902]
Stumbled upon this while migrating some code for selecting a batch of main eigenvectors to jax:

```python
import numpy as onp
import jax.numpy as jnp

B, M, N = 2, 3, 5
v, m = onp.random.rand(B, N), onp.random.rand(B, M, N)

def foo(np=None):
  i = np.argmax(v, axis=-1)[..., None, None]
  return np.take_along_axis(m, i, axis=-1)
print(foo(np=onp).shape) # => (2, 3, 1)
print(foo(np=jnp).shape) # => (2, 1, 1)
```

Jax returns something of a different shape. I also found that I could omit one of the dummy axes on `i`, which numpy does not accept (`m` and `i` must have the same ndim and compatible shapes) but jax does.

**Comment by cooijmanstim** [comment]
For the record, explicitly tiling `i` with `i = np.argmax(v, axis=-1)[..., None, None] * np.ones(m.shape, dtype=int)` works.

**Comment by mattjj** [comment]
Thanks for catching this!
